### PR TITLE
189 native end in web

### LINKED ISSUE #189: on iOS: Donation must finish inside web browser
## Acceptance Criteria

- [ ] 21.2  The collection of charitable donations must be done via a web site in Safari or an SMS
- [ ] Current app lets you get to the last step, then essentially does something like "your contribution must be finished in a web browser", you tap a button, it opens safari with a pre-authenticated URL and your contribution is on screen. You press finish and you're done.

### PR DESCRIPTION
**Pull request by Lepozepo**
Fixes #189 

## Creator

- [ ] Build relevant tests if any apply
- [ ] Ensure there are no new warnings
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [ ] Set a relevant reviewer

## Reviewer

- [ ] Run `test` and make sure all tests pass
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review function and form on Web
- [ ] Review new test logic

### CODE REVIEW

**Comment by Lepozepo** [comment]
This isn't ready yet, deep links had a few poorly documented caveats but should be good tomorrow now that they're working consistently and both users and contributions are resumable

**Comment by Lepozepo** [comment]
Oh, one more thing I need to add is an ENV for controlling the initial redirect, but here's the preview:

newspring

**Comment by Lepozepo** [comment]
@conrad-vanl this is ready to review

**Comment by conrad-vanl** [comment]
@Lepozepo looking good!!

One question: I notice this is using an in-app browser (which I guess you got to work - I couldn't get it to work before). However, I think we need to kick the user out to the actual browser, not the in-app browser.

**Comment by IsaacHardy** [comment]
@conrad-vanl Is the need to use the actual browser instead of `in-app-browser` because of Apple's guidelines or a design suggestion?

**Comment by conrad-vanl** [comment]
@IsaacHardy my understanding of apple guidelines:

> 21) Charities and contributions Apps that include the ability to make donations to recognized charitable organizations must be free The collection of donations must be done via a web site in Safari or an SMS

**Comment by conrad-vanl** [comment]
@IsaacHardy since apple's policy isn't very clear, we can always try it like it is, worse case they deny and we have to change it.

File: src/@ui/NativeWebRouter/DeepLinking.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  static contextTypes = {
+    router: PropTypes.shape({
+      history: PropTypes.shape({
+        push: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+  };
+
+  async componentDidMount() {
+    this.push('/give/now');
```
Comment (conrad-vanl) [inline src/@ui/NativeWebRouter/DeepLinking.js]: Why does this push to `/give/now` on mount?

**Comment by Lepozepo** [comment]
@conrad-vanl @IsaacHardy I replicated the flow for what we had in Holtzman (I may be missing some copy here and there) but it was definitely using an in app browser [here](https://github.com/NewSpring/Holtzman/blob/master/imports/util/inAppLink.js) with a fallback to safari (I think we get the fallback for free on expo but I need to double check).

File: src/@ui/NativeWebRouter/DeepLinking.js (hunk 1 above)
Comment (Lepozepo) [reply src/@ui/NativeWebRouter/DeepLinking.js]: Oh, that wasn't supposed to be there, lol

**Pull request merged**